contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,426 @@
+        }
+
+        AdjustTrove memory adjustTrove;
+
+        adjustTrove.maxFeePercentage = maxFeePercentage;
+        adjustTrove.withdrawAmount = withdrawAmount;
+        adjustTrove.depositAmount = getUint(getDepositId, depositAmount);
+        adjustTrove.borrowAmount = borrowAmount;
+        adjustTrove.repayAmount = getUint(getRepayId, repayAmount);
+        adjustTrove.isBorrow = borrowAmount > 0;
+
+        borrowerOperations.adjustTrove{value: depositAmount}(
+            adjustTrove.maxFeePercentage,
+            adjustTrove.withdrawAmount,
+            adjustTrove.borrowAmount,
+            adjustTrove.isBorrow,
+            upperHint,
+            lowerHint
+        );
+        
+        // Allow other spells to use the withdrawn collateral
+        setUint(setWithdrawId, withdrawAmount);
+
+        // Allow other spells to use the borrowed amount
+        setUint(setBorrowId, borrowAmount);
+
+        _eventName = "LogAdjust(address,uint256,uint256,uint256,uint256,uint256,uint256,uint256,uint256,uint256)";
+        _eventParam = abi.encode(msg.sender, maxFeePercentage, depositAmount, withdrawAmount, borrowAmount, repayAmount, getDepositId, setWithdrawId, getRepayId, setBorrowId);
+    }
+
+    /**
+     * @dev Withdraw remaining ETH balance from user's redeemed Trove to their DSA
+     * @param setId Optional storage slot to store the ETH claimed
+     * @notice Claim remaining collateral from Trove
+    */
+    function claimCollateralFromRedemption(uint setId) external returns(string memory _eventName, bytes memory _eventParam) {
+        uint amount = collateralSurplus.getCollateral(address(this));
+        borrowerOperations.claimCollateral();
+        setUint(setId, amount);
+
+        _eventName = "LogClaimCollateralFromRedemption(address,uint256,uint256)";
+        _eventParam = abi.encode(msg.sender, amount, setId);
+    }
+    /* End: Trove */
+
+    /* Begin: Stability Pool */
+
+    /**
+     * @dev Deposit LUSD into Stability Pool
+     * @notice Deposit LUSD into Stability Pool
+     * @param amount Amount of LUSD to deposit into Stability Pool
+     * @param frontendTag Address of the frontend to make this deposit against (determines the kickback rate of rewards)
+     * @param getDepositId Optional storage slot to retrieve the LUSD from
+     * @param setEthGainId Optional storage slot to store any ETH gains in
+     * @param setLqtyGainId Optional storage slot to store any ETH gains in
+    */
+    function stabilityDeposit(
+        uint amount,
+        address frontendTag,
+        uint getDepositId,
+        uint setEthGainId,
+        uint setLqtyGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        amount = getUint(getDepositId, amount);
+        uint ethGain = stabilityPool.getDepositorETHGain(address(this));
+        uint lqtyGain = stabilityPool.getDepositorLQTYGain(address(this));
```

Comment:
Because of the way LQTY rewards work, the returned value from `getDepositorLQTYGain` will be usually be slightly out of date, and slightly less than the actual LQTY paid out when the depositor does `provideToSP`.

How out of date it is depends on how frequently `StabilityPool` actions occur (by any depositor), but there's usually a 10-30 minute lag between them, and so `lqtyGain` will be a little low.

Not sure how much it matters here if that stored value is slightly less than the actual LQTY paid out.